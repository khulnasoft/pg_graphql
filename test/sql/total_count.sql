begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );


    insert into public.account(email)
    values
        ('a@x.com'),
        ('b@x.com');


    -- Should fail. totalCount not enabled
    select graphql.resolve($$
    {
      accountCollection {
        totalCount
        edges {
            cursor
        }
      }
    }
    $$);

    -- Enable totalCount
    comment on table account is e'@graphql({"totalCount": {"enabled": true}})';

    -- Should work. totalCount is enabled
    select graphql.resolve($$
    {
      accountCollection {
        totalCount
        edges {
            cursor
        }
      }
    }
    $$);

rollback;
