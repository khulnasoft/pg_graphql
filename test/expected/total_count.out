begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    insert into public.account(email)
    values
        ('a@x.com'),
        ('b@x.com');
    -- Should fail. totalCount not enabled
    select graphql.resolve($$
    {
      accountCollection {
        totalCount
        edges {
            cursor
        }
      }
    }
    $$);
                                resolve                                 
------------------------------------------------------------------------
 {"data": null, "errors": [{"message": "unknown field in connection"}]}
(1 row)

    -- Enable totalCount
    comment on table account is e'@graphql({"totalCount": {"enabled": true}})';
    -- Should work. totalCount is enabled
    select graphql.resolve($$
    {
      accountCollection {
        totalCount
        edges {
            cursor
        }
      }
    }
    $$);
                                                resolve                                                
-------------------------------------------------------------------------------------------------------
 {"data": {"accountCollection": {"edges": [{"cursor": "WzFd"}, {"cursor": "WzJd"}], "totalCount": 2}}}
(1 row)

rollback;
